lib/service.js

```diff
@@ -139,49 +139,36 @@ function processService(serviceMethod, env) {
         }
     }
 
-    function verifyCredentials(cfg, params) {
-        function doVerification(verify) {
-            return new Promise((resolve, reject) => {
-                function legacyCallback(e, result) {
-                    if (e) {
-                        return reject(e);
-                    }
-                    resolve(result);
-                }
-                var callScope = new EventEmitter();
-                callScope.logger = new ComponentLogger();
-                const result = verify.call(callScope, cfg, legacyCallback);
+    async function verifyCredentials(cfg, params) {
+        const verify = await compReader.loadVerifyCredentials();
 
-                if (result) {
-                    resolve(result);
-                }
-            });
-        }
+        const caller = new EventEmitter();
+        caller.logger = new ComponentLogger();
 
-        /**
-         * In will allow developers to return Promise.resolve(ANYTHING) in verifyCredentials.
-         */
-        function toVerifyCredentialsResponse(result) {
-            if (!_.has(result, 'verified')) {
-                return {
-                    verified: true
-                };
-            }
+        let response;
+        let promiseResponse;
 
-            return result;
-        }
-
-        function error(e) {
+        try {
+            response = await verify.call(caller, cfg, (e, verification) => {
+                if (e) {
```

Comment:
I'm not sure that callback processing will work in way you expected.
Please, add additional unit tests with both return and callback to ensure

Reply:
Will do!